civic_scraper/platforms/civic_clerk/site.py

```diff
@@ -0,0 +1,106 @@
+import re
+
+from datetime import datetime
+from io import StringIO
+from lxml import etree
+from requests import Session
+from pathlib import Path
+from urllib.parse import urlparse
+
+import civic_scraper
+from civic_scraper import base
+from civic_scraper.base.asset import Asset, AssetCollection
+from civic_scraper.base.cache import Cache
+
+
+class CivicClerkSite(base.Site):
+    def __init__(self, url, place=None, state_or_province=None, cache=Cache()):
+        self.url = url
+        self.base_url = 'https://' + urlparse(url).netloc
+        self.civicclerk_instance = urlparse(url).netloc.split('.')[0]
+        self.place = place
+        self.state_or_province = state_or_province
+        self.cache = cache
+
+    def create_asset(self, asset, committee_name, meeting_datetime, meeting_id):
+        asset_url, asset_name = asset
+        asset_type = 'Meeting'
+        full_asset_url = self.base_url + asset_url[2:]
+
+        e = {'url': full_asset_url,
+             'asset_name': asset_name,
+             'committee_name': committee_name,
+             'place': None,
+             'state_or_province': None,
+             'asset_type': asset_type,
+             'meeting_date': meeting_datetime.date(),
+             'meeting_time': meeting_datetime.time(),
+             'meeting_id': meeting_id,
+             'scraped_by': f'civic-scraper_{civic_scraper.__version__}',
+             'content_type': 'txt',
+             'content_length': None,
+            }
+        return Asset(**e)
+
+    def get_meeting_id(self, event):
+        link = event.xpath("./td[1]//a")[0]
+        href = link.attrib['href']
+        pattern = '.*?\((?P<id>.*?),.*'
+        match = re.match(pattern, href)
+        return 'civicclerk_{}_{}'.format(self.civicclerk_instance, match.group('id'))
+
+    def get_agenda_items(self, text, parser, session):
+        event_tree = etree.parse(StringIO(text), parser)
+        iframe = event_tree.xpath("//iframe")
+        if iframe:
```

Comment:
when would there not be an iframe? Add a comment here with an example url.

Reply:
I added this when I was first encountering the 500 errors; my expectation is that if things run as expected, there will always be an iframe. I ran the tests now and I'm still getting those 500 errors. I'm going to address the comments I can, then investigate a bit. I might reach out to pair tomorrow if I can't figure out where the errors are coming from